engine/robot.py

```diff
@@ -127,9 +128,10 @@ def move_to_target_node(self, target, allowed_dist_error, database):
 
         # location error (in meters)
         distance_away = self.calculate_dist(target, predicted_state)
+        self.robot_state.goal_location = target
         self.loc_pid_x.reset_integral()
         self.loc_pid_y.reset_integral()
-        while distance_away > allowed_dist_error:
+        while (distance_away > allowed_dist_error) and not (self.robot_state.phase == Phase.AVOID_OBSTACLE):
```

Comment:
Hmm just want to check my understanding here with the goal_location and our unvisited waypoints:

Lets say you were trying to move to target node X out of the waypoints [X, Y, Z]. You are still super far away from X but bam theres an obstacle in the way 🪨 . So the function breaks out of this while loop bc we don't want to traverse anymore we want to avoid obstacle (assuming the tracking function changes the phase here). However, at the conclusion of the `move_to_target_node` function we pop the leftmost unvisited waypoint (assuming the waypoint was fully visited if the function terminates), which is X but X was not fully visited. Ideally after we exit obstacle avoidance we would want to go back to moving to X. Can we assume visited X will be fully handled by obstacle avoidance by the time we return back to our traversal phase since we set the goal_location to that node aka `target`?

Reply:
This makes sense

Reply:
noice